### DISCUSSION TITLE
how do I provide extension description in for default extensions?

### DISCUSSION THREAD

**Discussion by softprops** [duckdb/duckdb#12037] [Q&A]
the interface for extensions is pretty bare. they can return a name and register a load hook

I like how duckdb is introspective with respect to extensions() but I can't find a way to provide a description that I see with default extensions. Can someone help? example description below

```
 from duckdb_extensions() limit 1;
┌────────────────┬─────────┬───────────┬──────────────┬──────────────────────────────────────────────────────────────┬───────────┬───────────────────┐
│ extension_name │ loaded  │ installed │ install_path │                         description                          │  aliases  │ extension_version │
│    varchar     │ boolean │  boolean  │   varchar    │                           varchar                            │ varchar[] │      varchar      │
├────────────────┼─────────┼───────────┼──────────────┼──────────────────────────────────────────────────────────────┼───────────┼───────────────────┤
│ arrow          │ false   │ false     │              │ A zero-copy data integration between Apache Arrow and DuckDB │ []        │                   │
└────────────────┴─────────┴───────────┴──────────────┴──────────────────────────────────────────────────────────────┴───────────┴───────────────────┘
```

**Comment by softprops** [comment]
NB [here](https://github.com/duckdb/duckdb/blob/d4c6e6713dbb0c682e3242cb173f5a7af1366448/src/main/extension/extension_helper.cpp#L103-L125) is where they are defined for default extensions

**Comment by carlopi** [comment]
Hi! We are looking into some improvements, and this was on the list of things that would be nice to add.

As of now, this is not currently possible, but stay tuned since this might change.
